pkg/R/DataFrame.R

```diff
@@ -774,21 +774,85 @@ setMethod("foreachPartition",
           })
 
 
-############################## DSL ##################################
+############################## SELECT ##################################
+
+getColumn <- function(x, c) {
+  column(callJMethod(x@sdf, "col", c))
+}
 
 setMethod("$", signature(x = "DataFrame"),
           function(x, name) {
-            column(callJMethod(x@sdf, "col", name))
+            getColumn(x, name)
+          })
+
+setMethod("$<-", signature(x = "DataFrame"),
+          function(x, name, value) {
+            stopifnot(class(value) == "Column")
+            cols <- columns(x)
+            if (name %in% cols) {
+              cols <- lapply(cols, function(c) {
+                if (c == name) {
+                  alias(value, name)
```

Comment:
Also, it would be **awesome** if we could assign `NULL` to a column name with this operator and have it drop the column from the DataFrame. This would mirror current `data.frame` behavior in R and the ability to easily drop a column of out of a Spark DataFrame would come in very handy.